test/dynamo/test_triton_kernels.py

```diff
@@ -1357,8 +1357,7 @@ def kernel_with_label(
                 "n_elements": 4,
                 "BLOCK_SIZE": 4,
             },
-            # TODO(oulgen): While loops not implemented yet
-            ["in_ptr0", "in_ptr1", "out_ptr"],
+            ["out_ptr"],
```

Comment:
All `add_4_times_kernel`s I see in this file have for-loops. I'm not sure which one exactly is used here, which, tbh, seems not great. I see that the one in `triton_utils.py` has  a while loop in it, so this PR is good. But do we want to do some name cleanup (and maybe even consolidate the kernels in one place, although I remember there was an issue with importing).

Reply:
This one is global scope, so it uses the one in `triton_utils.py`. That one has both while and for loop.

The others are defined in local scope which is used in their respective functions.

I define throwaway kernels in their local scope since we dont need them again. If a kernel needs to call another kernel, triton has a scope bug so they need to be defined globally.